test/testdata/rbi/redefine_superclass_payload.rb

```diff
@@ -0,0 +1,5 @@
+# typed: true
+
+class NewParent; end
+class Complex < NewParent
```

Comment:
I'm concerned about people doing really wrong things inside shims that we may not be able to control.

Would it make sense to allow this only based on a specific flag or sigil?

```rb
# typed: true
# allow-payload-parent-override: true
```

So we can at least set up a rule to review those changes?
```